src/cluster/placement/types.go

```diff
@@ -209,126 +209,54 @@ type Watch interface {
 	Close()
 }
 
-// DoneFn is called when caller is done using the resource.
-type DoneFn func()
-
-// StagedPlacementWatcher watches for updates to staged placement.
-type StagedPlacementWatcher interface {
+// Watcher watches for updates of the placement.
+type Watcher interface {
```

Comment:
What's the distinction between this and `Watch` above? Seems similar on face?

Reply:
There is a subtle distinction: the above requires clients to listen to notification channel if they want to react to placement changes -- more code on client side. This one invokes the callback function (if provided by the client via options) when placement changes -- less code on client side. 

The above one actually doesn't seem to be used at all, let me see if I can remove it.

Reply:
Hmm, I think it's better to keep it as it can watch both staged and non-staged placements but not gracefully -- it uses IsStaged option so that code using it must be sure about what kind of protobuf type is stored in etcd.

Reply:
Not having a way to subscribe to changes makes some unneccessary calls in aggregator that is in the critical path, it would be great to have a method to subscribe to changes, too - even if it's not in use by this PR yet.